coalib/misc/Exceptions.py

```diff
@@ -20,10 +15,10 @@ def get_exitcode(exception, log_printer=None):
         exitcode = exception.code
     elif isinstance(exception, VersionConflict):
         log_message = Constants.VERSION_CONFLICT_MESSAGE % str(exception.req)
-        log_printer.log_exception(log_message, exception)
+        LogPrinterMixin().log_exception(log_message, exception)
```

Comment:
I wonder if it makes sense to split this PR up into multiple commits or even PRs, which are smaller and therefore easier to review. @userzimmermann WDYT?